components/dashboard/src/start/WorkspaceLogs.tsx

```diff
@@ -4,8 +4,81 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
-export default function WorkspaceLogs() {
-  return <pre className="m-6 p-4 h-72 w-11/12 lg:w-3/5 flex-shrink-0 rounded-xl bg-gray-100">
-    ... logs ...
-  </pre>;
+import EventEmitter from 'events';
+import React from 'react';
+import { Terminal, ITerminalOptions, ITheme } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit'
+import 'xterm/css/xterm.css';
+import { DisposableCollection } from '@gitpod/gitpod-protocol';
+
+export interface WorkspaceLogsProps {
+  logsEmitter: EventEmitter;
+}
+
+export interface WorkspaceLogsState {
+}
+
+export default class WorkspaceLogs extends React.Component<WorkspaceLogsProps, WorkspaceLogsState> {
+  protected xTermParentRef: React.RefObject<HTMLDivElement>;
+  protected terminal: Terminal | undefined;
+  protected fitAddon: FitAddon | undefined;
+  
+  constructor(props: WorkspaceLogsProps) {
+    super(props);
+    this.xTermParentRef = React.createRef();
+  }
+
+  private readonly toDispose = new DisposableCollection();
+  componentDidMount() {
+    const element = this.xTermParentRef.current;
+    if (element === null) {
+      return;
+    }
+    const theme: ITheme = {
+      // background: '#F5F5F4',
+    };
+    const options: ITerminalOptions = {
```

Comment:
**question**: Is there an option to set the width here? It could be fun and a bit futuristic to use the `w-96` / `24rem` / `384px` width here for the terminal.

Reply:
I think you can just specify the number of rows and columns the Terminal should have: https://xtermjs.org/docs/api/terminal/interfaces/iterminaloptions/